mintpy/save_geodmod.py

```diff
@@ -0,0 +1,166 @@
+    # initiate dict
+    rsc = dict()
+    rsc['FILE_DIR'] = 'lxr/Iraq/afterslip_example'
+    rsc['FILE_LENGTH'] = meta["FILE_LENGTH"]
+    rsc['WIDTH'] = meta["WIDTH"]
+    rsc['XMIN'] = 0
+    rsc['XMAX'] = int(meta["WIDTH"]) - 1
+    rsc['YMIN'] = 0
+    rsc['YMAX'] = int(meta["FILE_LENGTH"]) - 1
+    rsc['X_FIRST'] = float(meta["X_FIRST"])
+    rsc['Y_FIRST'] = float(meta["Y_FIRST"])
+    rsc['X_STEP'] = float(meta["X_STEP"])
+    rsc['Y_STEP'] = float(meta["Y_STEP"])
+    rsc['X_UNIT'] = 'degrees'
+    rsc['Y_UNIT'] = 'degrees'
+    rsc['RLOOKS'] = meta["RLOOKS"]
+    rsc['ALOOKS'] = meta["ALOOKS"]
+    rsc['Z_OFFSET'] = 0
+    rsc['Z_SCALE'] = 1
+    rsc['PROJECTION'] = 'LATLON'
+    rsc['DATE12'] = '111111-222222'
+    # write rsc file
+    writefile.write_roipac_rsc(rsc, out_file, print_msg=True)
+    return out_file 
+
+######################################################################################
+def main(iargs=None):
+    inps = cmd_line_parse(iargs)
+    
+   
+    if inps.Type=='unw':
+        #  geocode timeseries**.h5 file and get the deformation field of two time periods
+        atr_asc = inps.file[0]
+        os.chdir(os.path.dirname(atr_asc))
+    
+        asc_args = [atr_asc, '-b',inps.SNWE, '-y',inps.latStep, '-x',inps.lonStep]
+        print("geocode.py", asc_args)
+        args_str = format_args(asc_args)
+        mintpy.geocode.main(args_str.split())
+    
+        _, filename, extension = get_path_com(atr_asc)
+        asct_args = ['geo_'+filename+extension, inps.dset[0], '-o', inps.outfile]
+        print("save_roipac.py", asct_args)
+        asct_str = format_args(asct_args)
+        mintpy.save_roipac.main(asct_str.split())
+    elif inps.Type=='cor':
+        #  geocode ifgramStack.h5 file and get the coherence of two time periods
+        atr_asc = inps.file[0]
+        os.chdir(os.path.dirname(atr_asc))
+        
+        asc_args = [atr_asc, '-b',inps.SNWE, '-y',inps.latStep, '-x',inps.lonStep]
+        print("geocode.py", asc_args)
+        args_str = format_args(asc_args)
+        mintpy.geocode.main(args_str.split())
+    
+        _, filename, extension = get_path_com(atr_asc)
+        if filename=='ifgramStack':
+            asct_args = ['geo_'+filename+extension, 'coherence-'+inps.dset[0], '-o', inps.outfile]
+        else:        
+            asct_args = ['geo_'+filename+extension, '-o', inps.outfile]
+        print("save_roipac.py", asct_args)
+        asct_str = format_args(asct_args)
+        mintpy.save_roipac.main(asct_str.split())
+    else:
+        #  geocode geometryRadar.h5 file and get the dem
+        atr_asc = inps.file[0]
+        os.chdir(os.path.dirname(atr_asc))
+        
+        asc_args = [atr_asc, '-b',inps.SNWE, '-y',inps.latStep, '-x',inps.lonStep]
+        print("geocode.py", asc_args)
+        args_str = format_args(asc_args)
+        mintpy.geocode.main(args_str.split())
+    
+        _, filename, extension = get_path_com(atr_asc)
```

Comment:
use `filename, extension = get_path_com(atr_asc)[1:3]` instead?

Reply:
Thank you for your suggestions and I modified this part.